src/Bicep.Core/Parser/Parser.cs

```diff
@@ -297,25 +303,112 @@ private SyntaxBase ParenthesizedExpression()
 
         private SyntaxBase FunctionCallOrVariableAccess()
         {
-            var identifier = this.Identifier(b => b.ExpectedFunctionName());
+            if (IsFunctionCall())
```

Comment:
I feel like it would be clearer to consume the identifier directly inside `FunctionCallOrVariableAccess`, and then make the decision whether to generate FunctionCallSyntax or VariableAccessSyntax depending on whether the next token is a `(` or not. That avoids the need to look ahead, and as Marcin pointed out, will probably allow us to give a better error if we don't see an identifier.